app/controllers/api/articles_controller.rb

```diff
@@ -1,6 +1,10 @@
 class Api::ArticlesController < ApplicationController
   def index
-    articles = Article.all
+    articles = if params[:category_name]
+                 Article.where(category_name: params[:category_name])
+               else
+                 Article.all
+               end
```

Comment:
you can extract that if statement into a private function. More short functions is better than few big ones.